External/RapidJSON/CMakeLists.txt

```diff
@@ -1,3 +1,9 @@
+# Check if the RapidJSON target has already been defined
+if (TARGET RapidJSON)
+  message(AUTHOR_WARNING "RapidJSON target already defined, skipping")
```

Comment:
Should this just be a `STATUS` message? People who deliberately predefine these targets probably won't want these warnings printed every time.

Reply:
Maybe.  I wanted to see when it was happening and part of making it an "AUTHOR_WARNING" instead of just "WARNING" is that it will only appear if you are building in dev mode (building with -Wno-dev will suppress it).  I also wanted to make sure it was clear if this were to happen by accident.